lib/ast_parser.ex

```diff
@@ -145,6 +146,22 @@ defmodule ExcellentMigrations.AstParser do
 
   defp detect_column_modified(_), do: []
 
+  defp detect_column_added_with_reference(
+         {:add, location, [_, {:references, _, [_column, options]}]}
+       ) do
```

Comment:
I wonder if it would be possible to refactor it to have one place that detects both add & modify:
```suggestion
  defp detect_column_reference_added(
         {fun_name, location, [_, {:references, _, [_column, options]}]}
       ) when fun_name in ["add", "modify"] do
```